FreeRTOS/FreeRTOS-Kernel
# Move configASSERT default definition above before including portable.h.

Move configASSERT default definition above before including portable.h.

Description
-----------

As #1170 mentioned, for RP2040 port, there is a compile error if users don't define configASSERT in FreeRTOSConfig.h.
In this PR, we move the configASSERT default definition above before including `portable.h` to solve this problem in a generic method.

Test Steps
-----------

1. Remove configASSERT definition in FreeRTOS/Test/Target/boards/pico/FreeRTOSConfig.h.
2. Updating Kernel submodule pointer to the working forked branch.
3. Compile RP2040 target testing load.
4. Pass compilation.
```
export PICO_SDK_PATH=$PATH_TO_PICO_SDK
rm -fr $PATH_TO_FREERTOS/FreeRTOS/FreeRTOS/Test/Target/boards/pico/build
cd $PATH_TO_FREERTOS/FreeRTOS/FreeRTOS/Test/Target/boards/pico
export FREERTOS_KERNEL_PATH=$PATH_TO_FREERTOS/FreeRTOS/FreeRTOS/Source
cmake -S . -B build
cd build
make -j
```

Checklist:
----------



Related Issue
-----------



By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
